extensions/data-plane/data-plane-http/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceTest.java

```diff
@@ -84,38 +66,23 @@ void verifyRequest_withNameAndQueryParams() {
 
         source.openPartStream();
 
-        var requests = interceptor.getRequests();
-        assertThat(requests).hasSize(1);
-        assertThat(requests.get(0)).satisfies(request -> {
-            assertThat(request.url()).hasToString(TEST_ENDPOINT + "/testfile?foo=bar&hello=world");
-            assertThat(request.method()).isEqualTo("GET");
-        });
+        var request = interceptor.getInterceptedRequest();
+        assertThat(request.url()).hasToString(TEST_ENDPOINT + "/testfile?foo=bar&hello=world");
+        assertThat(request.method()).isEqualTo("GET");
     }
 
     @Test
     void verifyRequest_body() {
         var interceptor = new CustomInterceptor();
         var json = "{ \"foo\" : \"bar\" }";
-        var requestBody = RequestBody.create(json, MediaType.get("application/json"));
-        var source = defaultBuilder(interceptor).method("POST").requestBody(requestBody).build();
+        var source = defaultBuilder(interceptor).method("POST").requestBody(MediaType.get("application/json"), json).build();
 
         source.openPartStream();
 
-        var requests = interceptor.getRequests();
-        assertThat(requests).hasSize(1);
-        assertThat(requests.get(0)).satisfies(request -> {
-            assertThat(request.url()).hasToString(TEST_ENDPOINT + "/");
-            assertThat(request.method()).isEqualTo("POST");
-            try {
-                // check request body
-                var sink = Okio.sink(new ByteArrayOutputStream());
-                var bufferedSink = Okio.buffer(sink);
-                Objects.requireNonNull(request.body()).writeTo(bufferedSink);
-                assertThat(bufferedSink.getBuffer().readUtf8()).isEqualTo(json);
-            } catch (IOException e) {
-                throw new AssertionError("Failed to write body");
-            }
-        });
+        var request = interceptor.getInterceptedRequest();
+        assertThat(request.url()).hasToString(TEST_ENDPOINT + "/");
+        assertThat(request.method()).isEqualTo("POST");
+        verifyRequestBody(request, json);
```

Comment:
This hides the assertion, maybe would be more expressive to have a method that extracts body from request:
```
    private String getBody(Request request) {
        try {
            var sink = Okio.sink(new ByteArrayOutputStream());
            var bufferedSink = Okio.buffer(sink);
            RequestBody body = request.body();
            Objects.requireNonNull(body).writeTo(bufferedSink);
            return bufferedSink.getBuffer().readUtf8();
        } catch (IOException e) {
            throw new AssertionError(e);
        }
    }
```

and then leave the assertion in the test method:
```
assertThat(getBody(request)).isEqualTo(json);
```